app.R

```diff
@@ -1,4 +1,345 @@
-# author: Christopher Alexander, Jennifer Hoang, Michelle Wang, Thea Wenxin
-# date: 2022-03-01
+# author: Christopher Alexander, Jennifer Hoang, Michelle Wang, Wenxin Xiang
+# date: 2022-03-15
 
-pass
+library(dash)
+library(dashHtmlComponents)
+library(dashBootstrapComponents)
+library(dashCoreComponents)
+library(ggplot2)
+library(plotly)
+library(purrr)
+library(readr)
+library(dplyr)
+
+# Read data ------
+
+df <- read_csv("https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-01-21/spotify_songs.csv")
```

Comment:
Could change it to local data. As URL request from heroku will again take time.

Reply:
I will need to test it out on my local, last time it cannot read data from local directory.